scripts/quests/otherAreas/A_Generous_General.lua

```diff
@@ -0,0 +1,290 @@
+                        pos.y <= 9.0 and
+                        pos.z <= -258.0
+                    then
+                        return 60
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [60] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player) -- begin quest
+                    elseif option == 1 then
+                        player:setVar('BstHeadGearQuest_Conquest', NextConquestTally()) -- if player declines they must wait until next conquest tally
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Faulpie'] =
+            {
+                onTrigger = function(player, npc)
+                    local prog = quest:getVar(player, 'Prog')
+                    local partsWait = quest:getVar(player, 'partsWait')
+                    if prog == 0 then
+                        return quest:progressEvent(770) -- tells player what items are needed
+                    elseif prog == 1 then
+                        return quest:event(771):importantOnce() -- reminder to get buffalo hide and sheep leather or cancel quest
+                    elseif
+                        prog >= 2 and
+                        partsWait ~= 0 -- player is currently waiting for parts
+                    then
+                        if partsWait <= os.time() then
+                            return quest:progressEvent(775) -- ready to go
+                        else
+                            return quest:event(773):importantOnce() -- player must wait till vana midnight and zone(era only)
+                        end
+                    elseif prog == 3 then
+                        return quest:event(774):importantOnce() -- Get another part for 100000 gil or give up
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, { xi.item.BUFFALO_HIDE, xi.item.SQUARE_OF_SHEEP_LEATHER, { 'gil', 10000 } }) and
+                        quest:getVar(player, 'Prog') == 1
+                    then
+                        return quest:progressEvent(772)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [770] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+
+                [771] = function(player, csid, option, npc)
+                    if option == 100 then
+                        player:messageSpecial(southernSanDoriaID.text.QUEST_CANCELLED)
+                        quest:setVar(player, 'Prog', 0) -- reset progress
```

Comment:
There are other quest vars here that may not be cleaned up, call `quest:cleanup(player)`